### PR TITLE
Exclude unneeded files/folders from git packages

### PR DESCRIPTION
**Pull request by sgiehl**
replaces #9406

### CODE REVIEW

**Comment by tsteur** [comment]
Should we maybe add some more from here: https://github.com/piwik/piwik-package/blob/master/scripts/build-package.sh#L111 ? Some people seem to download ZIP from Github and this way it would contain similar content to the actual build.zip. Or maybe we would not even need it in the build-package anymore? That would be ideal

**Comment by sgiehl** [comment]
Sure, we could include some more files for exclusion, but I'm unsure which files....
But as long as the build-package doesn't use a git archive the files needs to be removed there aswell.

File: .gitattributes [hunk 1]
```diff
@@ -41,3 +41,14 @@
 
 # Custom for Visual Studio
 *.cs     diff=csharp
+
+
+# Files excluded from git packages
+tests/ export-ignore
```
Comment (tsteur) [inline .gitattributes]: For some reasons we also allow to have tests in `Test` directory which we should remove as well if these paths apply to nested directories as well? or do they only apply to directories in root directory? Maybe we need to define `/tests` or something similar to .gitignore?

Comment (tsteur) [inline .gitattributes]: does it automatically remove .git folder?

**Comment by tsteur** [comment]
Added a comment re absolute paths but otherwise seems ok to merge.

FYI: This also affects zip downloads on github. I presume we want to remove it from there as well so it's 👍  but mentioning it just in case we want to let users download everything from there @mattab 

We certainly should create a follow up issue re https://github.com/piwik/piwik-package/blob/master/scripts/build-package.sh#L111 so we have a unified logic there and the definitions only in one place.

**Comment by tsteur** [comment]
FYI: The branch points to 3.0 but should be now 3.x-dev

**Comment by mattab** [comment]
Hi @sgiehl - could you  resolve the merge conflict? Then Looks good to mereg

**Comment by sgiehl** [comment]
rebased it on latest `3.x-dev` branch.

**Pull request merged**